python/lsst/ap/association/assoc_afw_utils.py

```diff
@@ -0,0 +1,279 @@
+                alias_map = source_catalog.getSchema().getAliasMap()
+                alias_map.set('psFlux', 'base_PsfFlux_flux')
+                alias_map.set('psFluxErr', 'base_PsfFlux_fluxSigma')
+
+
+def convert_dia_source_to_asssoc_schema_and_calibrate(dia_sources,
+                                                      obj_ids=None,
+                                                      exposure=None):
+    """Create aliases in for the input source catalog schema and overwrite
+    values if requested.
+
+    Parameters
+    ----------
+    dia_sources : `lsst.afw.table.SourceCatalog`
+        Input source catalog to alias and overwrite values of.
+    obj_ids:  array-like of `int`
+        DIAObject ids to associated with the input DIASources. Overwrites
+        the current value in diaObjectId column.
+    exposure : `lsst.afw.image.Exposure`
+        Exposure to copy flux and filter information from.
+    """
+
+    output_dia_sources = afwTable.SourceCatog(
+        make_minimal_dia_source_schema())
+    output_dia_sources.reserve(len(dia_sources))
+    add_dia_source_aliases_to_catalog(dia_sources)
+    dia_source_schema = dia_sources.getSchema()
+
+    if exposure is None:
+        exp_dict = None
+    else:
+        exp_dict = get_ccd_visit_info_from_exposure(exposure)
+
+    for src_idx, source_record in enumerate(dia_sources):
+        if obj_ids is None:
+            obj_id = None
+        else:
+            obj_id = obj_ids[src_idx]
+        overwrite_dict = make_overwrite_dict(source_record, obj_id, exp_dict)
+        output_record = output_dia_sources.addNew()
+        copy_source_record(source_record,
+                           output_record,
+                           dia_source_schema,
+                           overwrite_dict)
+    return output_dia_sources
+
+
+def make_overwrite_dict(source_record, obj_id=None, exp_dict=None):
+    """Create a dictionary of values to overwrite the values stored in an
+    afw.table or when storing values in a DB.
+
+    Parameters
+    ----------
+    source_record : `lsst.afw.table.SourceRecord`
+        Input source catalog to modify values of if needed.
+    obj_id : `int` (optional)
+        DIAObject id to overwrite.
+    exp_dict : `dict`
+        Dictionary of exposure properties for this source_catalog.
+
+    Returns
+    -------
+    overwrite_dict : `dict`
+        Dictionary of keys and values to overwrite for this source record.
+    """
+    overwrite_dict = {}
+    if obj_id is not None:
+        overwrite_dict['diaObjectId'] = int(obj_id)
+    if exp_dict is not None:
+        psFlux = source_record['psFlux']
+        psFluxErr = source_record['psFluxErr']
+
+        overwrite_dict['psFlux'] = psFlux / exp_dict['fluxZeroPoint']
+        overwrite_dict['psFluxErr'] = np.sqrt(
+            (psFluxErr / exp_dict['fluxZeroPoint']) ** 2 +
+            (psFlux * exp_dict['fluxZeroPointErr'] /
+             exp_dict['fluxZeroPoint'] ** 2) ** 2)
```

Comment:
I don't like that we're coding the error calculation here explicitly. Call something from `Calib` or `PhotoCalib` instead. For example, `PhotoCalib.instFluxToMaggies(flux, fluxErr)` will give you the calibrated flux and fluxErr.

Reply:
Do calexps or other exposure like objects have a PhotoCalib object? The calib object stored doesn't have an option to convert to maggies. Is there an example of how I can use a PhotoCalib object with an exposure and a catalog?

Reply:
A more future-proof approach would be to create a `PhotoCalib` object when you read in the Exposure and extract the `Calib` and pass it around, instead of the zero-point. This should do it:

```
fluxMag0 = calib.getFluxMag0()
photoCalib = afwImage.PhotoCalib(1.0/fluxMag0[0], fluxMag0[1]/fluxMag0[0]**2, bbox)
```

That way, once we replace Calib with PhotoCalib (DM-10153), we just have to delete two lines.

Oh, please leave a TODO as well.